Make the C++ standard configurable (#2319)
nest/nest-simulator: CMakeLists.txt

set( with-optimize ON CACHE STRING "Enable user defined optimizations [default=OFF (uses '-O2')]. When ON, '-O3' is used. Separate multiple flags by ';'." )
set( with-warning ON CACHE STRING "Enable user defined warnings [default=ON (uses '-Wall')]. Separate  multiple flags by ';'." )
set( with-debug OFF CACHE STRING "Enable user defined debug flags [default=OFF]. When ON, '-g' is used. Separate  multiple flags by ';'." )

> set( with-std OFF CACHE STRING "User defined C++ standard [default='c++11']." )

Reviewer: ```suggestion
set( with-std OFF CACHE STRING "C++ standard to use for compilation [default='c++11']." )
```
We don't really allow users to define their own C++ standards ;).
